tests/load/prometheus.go

```diff
@@ -58,23 +58,21 @@ func (s *MetricsServer) Start() (runError <-chan error, err error) {
 		ReadTimeout:       time.Second,
 	}
 
-	runErrorBiDirectional := make(chan error)
-	runError = runErrorBiDirectional
-	ready := make(chan struct{})
+	errChan := make(chan error)
 	go func() {
-		close(ready)
```

Comment:
It's not clear to me why we would create a channel just to wait for the goroutine to close it. It seems this is to ensure the goroutine has started before we return? Maybe this is trying to guarantee that we can reach the server as soon as this function returns, but that would still be race-y given we close the channel before starting the server in the goroutine.